packages/camera/camera/android/src/main/java/io/flutter/plugins/camera/features/CameraFeature.java

```diff
@@ -0,0 +1,56 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package io.flutter.plugins.camera.features;
+
+import android.hardware.camera2.CaptureRequest;
+import androidx.annotation.NonNull;
+import io.flutter.plugins.camera.CameraProperties;
+
+/**
+ * An interface describing a feature in the camera. This holds a setting value of type T and must
+ * implement a means to check if this setting is supported by the current camera properties. It also
+ * must implement a builder update method which will update a given capture request builder for this
+ * feature's current setting value.
+ *
+ * @param <T>
+ */
+public abstract class CameraFeature<T> {
+  protected final CameraProperties cameraProperties;
+
+  protected CameraFeature(@NonNull CameraProperties cameraProperties) {
+    this.cameraProperties = cameraProperties;
+  }
+
+  /** Debug name for this feature. */
+  public abstract String getDebugName();
+
+  /**
+   * Get the current value of this feature's setting.
```

Comment:
What is the expected behavior of get/set if `checkIsSupported` returns false? That would be good to document.